index.js

```diff
@@ -12,3 +13,15 @@ const pagination = document.querySelector('[data-js="pagination"]');
 const maxPage = 1;
 const page = 1;
 const searchQuery = "";
+
+async function fetchCharacters() {
+  try {
+    const response = await fetch("https://rickandmortyapi.com/api/character");
+    const { results } = await response.json();
+    results.forEach(createCharacterCard);
+  } catch (error) {
+    console.error("An error occurred while fetching characters:", error);
+  }
+}
```

Comment:
Nice `try` & `catch` block!
Please have a look at PR1. You are supposed to return the `li`-element from the `createCharacterCard` function and append it inside the `fetchCharacters` function right after you have created the card.

Additionally, you should always clear the `cardContainer` ( ` cardContainer.innerHTML = "";`) before creating new cards.